src/Compilers/Core/Portable/Collections/TemporaryArray`1.cs

```diff
@@ -351,6 +351,58 @@ public void ReverseContents()
             }
         }
 
+        public void Sort(Comparison<T> compare)
+        {
+            if (_builder is not null)
+            {
+                _builder.Sort(compare);
+                return;
+            }
+
+            switch (_count)
+            {
+                case <= 1:
+                    return;
+                case 2:
+                    if (compare(_item0, _item1) > 0)
+                    {
+                        (_item0, _item1) = (_item1, _item0);
+                    }
+                    return;
+                case 3:
+                    if (compare(_item0, _item1) > 0)
+                        (_item0, _item1) = (_item1, _item0);
+
+                    if (compare(_item1, _item2) > 0)
+                        (_item1, _item2) = (_item2, _item1);
+
+                    if (compare(_item0, _item1) > 0)
```

Comment:
>if (compare(_item0, _item1) > 0) [](http://example.com/codeflow?start=20&length=32)

It looks like there is no need for this comparison, unless we entered the previous `if`